### PR TITLE
Add network policy

### PR DESCRIPTION
**Pull request by glitchcrab**
towards giantswarm/giantswarm/issues/4383

external egress is required.

### CODE REVIEW

File: helm/app-operator-chart/templates/network-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  namespace: {{ .Values.namespace }}
+  name: {{ .Values.name }}-network-policy
```
Comment (kopiczko) [inline helm/app-operator-chart/templates/network-policy.yaml]: Isn't it good to add `app` label also here?

Comment (corest) [reply helm/app-operator-chart/templates/network-policy.yaml]: Even more, like this https://github.com/giantswarm/net-exporter/blob/master/helm/net-exporter/templates/np.yaml#L7-L8

File: helm/app-operator-chart/templates/network-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    matchLabels:
+      app: {{ .Values.name }}
+  ingress:
+  - ports:
+    - port: {{ .Values.port }}
+      protocol: {{ .Values.protocol }}
+  egress:
+  - to:
+    - ipBlock:
+        cidr: 0.0.0.0/0
```
Comment (tomahawk28) [inline helm/app-operator-chart/templates/network-policy.yaml]: I'm sorry to ask, but does it basically open to all address?

Comment (corest) [reply helm/app-operator-chart/templates/network-policy.yaml]: Yes.
Also, we're usually doing just `{}`, so maybe let's stick to single style

**Comment by rossf7** [comment]
@glitchcrab I think this is done. Can this PR be closed?

**Comment by glitchcrab** [comment]
@rossf7 it looks that way - i'll rebase and re-deploy to test again

**Pull request merged**